enclave/src/main/java/org/hyperledger/besu/enclave/Enclave.java

```diff
@@ -26,101 +28,117 @@
+    clientOptions.setDefaultPort(enclaveUri.getPort());
+
+    vertx = Vertx.vertx();
+    this.client = vertx.createHttpClient(clientOptions);
   }
 
-  public boolean upCheck() throws IOException {
-    final String url = enclaveUri.resolve("/upcheck").toString();
-    final Request request = new Request.Builder().url(url).get().build();
+  public boolean upCheck() throws ExecutionException, InterruptedException {
+    final CompletableFuture<Boolean> result = new CompletableFuture<>();
+    final HttpClientRequest request = client.request(HttpMethod.GET, "/upcheck").setTimeout(1000);
+    request.end(asyncResult -> result.complete(asyncResult.succeeded()));
 
-    try (Response response = client.newCall(request).execute()) {
-      return response.isSuccessful();
-    } catch (IOException e) {
-      LOG.error("Enclave failed to execute upcheck");
-      throw new IOException("Failed to perform upcheck", e);
-    }
+    return result.get();
   }
 
   public SendResponse send(final SendRequest content) {
-    return executePost(buildPostRequest(JSON, content, "/send"), SendResponse.class);
+    return postRequest(JSON, content, "/send", SendResponse.class);
   }
 
   public ReceiveResponse receive(final ReceiveRequest content) {
-    return executePost(buildPostRequest(ORION, content, "/receive"), ReceiveResponse.class);
+    return postRequest(ORION, content, "/receive", ReceiveResponse.class);
   }
 
   public PrivacyGroup createPrivacyGroup(final CreatePrivacyGroupRequest content) {
-    return executePost(buildPostRequest(JSON, content, "/createPrivacyGroup"), PrivacyGroup.class);
+    return postRequest(JSON, content, "/createPrivacyGroup", PrivacyGroup.class);
   }
 
   public String deletePrivacyGroup(final DeletePrivacyGroupRequest content) {
-    return executePost(buildPostRequest(JSON, content, "/deletePrivacyGroup"), String.class);
+    return postRequest(JSON, content, "/deletePrivacyGroup", String.class);
   }
 
   public PrivacyGroup[] findPrivacyGroup(final FindPrivacyGroupRequest content) {
-    Request request = buildPostRequest(JSON, content, "/findPrivacyGroup");
-    return executePost(request, PrivacyGroup[].class);
+    return postRequest(JSON, content, "/findPrivacyGroup", PrivacyGroup[].class);
   }
 
-  private Request buildPostRequest(
-      final MediaType mediaType, final Object content, final String endpoint) {
-    final RequestBody body;
+  private <T> T postRequest(
+      final AsciiString mediaType,
+      final Object content,
+      final String endpoint,
+      final Class<T> responseType) {
+
     try {
-      body = RequestBody.create(mediaType, objectMapper.writeValueAsString(content));
+      final CompletableFuture<T> result = new CompletableFuture<>();
+      final HttpClientRequest request = client.request(HttpMethod.POST, endpoint);
+      request.handler(response -> handleResponse(response, responseType, result));
+      request.putHeader("content-Type", mediaType);
```

Comment:
can use the io.vertx.core.http.HttpHeaders.CONTENT_TYPE constant

Reply:
done